app/components/Search/mazemapAutocomplete.tsx

```diff
@@ -1,112 +1,120 @@
-      });
-      this._isMounted = true;
-    }
+export const useMazemapAutocomplete = () => {
+  const [options, setOptions] = useState<SelectOption[]>([]);
+  const [fetching, setFetching] = useState(false);
+  const [mazemapSearchController, setMazemapSearchController] =
+    useState<MazemapSearchController>();
 
-    componentWillUnmount() {
-      this._isMounted = false;
+  useEffect(() => {
+    import('mazemap').then((mazemap) => {
+      setMazemapSearchController(
+        new mazemap.Search.SearchController({
+          campusid: 1,
+          rows: 10,
+          withpois: true,
+          withbuilding: false,
+          withtype: false,
+          withcampus: false,
+          resultsFormat: 'geojson',
+        })
+      );
+    });
+  }, []);
+
+  const onSearch = (query: string) => {
+    if (!query) {
+      return;
     }
 
-    handleSearch = (query: string): void => {
-      if (!query) {
-        return;
-      }
+    setFetching(true);
 
-      this.setState({
-        searching: true,
-      });
+    mazemapSearchController &&
+      mazemapSearchController
+        .search(query)
+        .then((results) => {
+          setOptions(
+            results.results.features.map((result) =>
+              mapRoomAndBuildingToSelectOption(
+                result.properties.dispPoiNames[0],
+                result.properties.dispBldNames[0],
+                result.properties.poiId
+              )
+            )
+          );
+        })
+        .catch(() => {})
+        .finally(() => setFetching(false));
+  };
 
-      if (this._isMounted) {
-        this.state.mazemapSearchController &&
-          this.state.mazemapSearchController
-            .search(query)
-            .then((results) => {
-              this.setState({
-                result: results.results.features.map((result) =>
-                  mapRoomAndBuildingToResult(
-                    result.properties.dispPoiNames[0],
-                    result.properties.dispBldNames[0],
-                    result.properties.poiId
-                  )
-                ),
-              });
-            })
-            .catch(() => {})
-            .finally(() =>
-              this.setState({
-                searching: false,
-              })
-            );
-      }
-    };
+  const debouncedOnSearch = debounce(onSearch, 100);
 
-    render() {
-      return (
-        <WrappedComponent
-          {...this.props}
-          options={this.state.result}
-          onSearch={debounce((query) => this.handleSearch(query), 300)}
-          fetching={this.state.searching}
-        />
-      );
-    }
+  return { options, onSearch: debouncedOnSearch, fetching };
+};
+
+const withMazemapAutocomplete = <P,>({
+  WrappedComponent,
+}: {
+  WrappedComponent: ComponentType<P & InjectedProps>;
+}) => {
+  const Component = (props: P) => {
+    const { options, onSearch, fetching } = useMazemapAutocomplete();
+
+    return (
+      <WrappedComponent
+        {...(props as P)}
```

Comment:
A bit confused here, is there a reason why typecasting is necessary here?

Reply:
Yes

Reply:
Actually no, I'll remove it